### PR TITLE
[SPARK-42460][CONNECT] Clean-up results in ClientE2ETestSuite

### PR DESCRIPTION
**Pull request by hvanhovell**
### What changes were proposed in this pull request?
Clean-up results in `ClientE2ETestSuite`.

### Why are the changes needed?
`ClientE2ETestSuite` is very noisy because we do not clean-up results. This makes testing a bit annoying.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
It is a test.

### CODE REVIEW

File: connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/ClientE2ETestSuite.scala [hunk 1]
```diff
@@ -32,40 +32,44 @@ class ClientE2ETestSuite extends RemoteSparkSession {
...
 
   test("spark result array") {
     val df = spark.sql("select val from (values ('Hello'), ('World')) as t(val)")
-    val result = df.collectResult()
-    assert(result.length == 2)
-    val array = result.toArray
-    assert(array.length == 2)
-    assert(array(0).getString(0) == "Hello")
-    assert(array(1).getString(0) == "World")
+    df.withResult { result =>
```
Comment (hvanhovell) [inline connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/ClientE2ETestSuite.scala:43]: we could also use df.collect() here.

**Comment by hvanhovell** [comment]
cc @zhenlineo @LuciferYang

**Review by HyukjinKwon — APPROVED** [review]

**Review by LuciferYang — APPROVED** [review]
fine to me

**Comment by LuciferYang** [comment]
https://github.com/apache/spark/blob/7ee8a32077b09cb847b6ac41cdc5067cf7bd83e9/connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/connect/client/util/RemoteSparkSession.scala#L155-L159

should we remove this `try catch`? This may cover up some problems, but this may be another pr

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by dongjoon-hyun** [comment]
According to your comment, is it okay for me to merge, @LuciferYang ?

**Comment by LuciferYang** [comment]
> According to your comment, is it okay for me to merge, @LuciferYang ?

ok to merge ~

**Comment by dongjoon-hyun** [comment]
Thanks, @LuciferYang . Merged to master/3.4.
Thank you, @hvanhovell , @HyukjinKwon , @LuciferYang .

**Pull request closed without merging**